common/src/main/java/com/microsoft/identity/common/internal/ui/webview/challengehandlers/SwitchBrowserUriHelper.kt

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.internal.ui.webview.challengehandlers
+
+import android.net.Uri
+import com.microsoft.identity.common.adal.internal.AuthenticationConstants.Broker
+import com.microsoft.identity.common.adal.internal.AuthenticationConstants.SWITCH_BROWSER
+import com.microsoft.identity.common.java.AuthenticationConstants.OAuth2
+import com.microsoft.identity.common.logging.Logger
+
+/**
+ * SwitchBrowserUriHelper is a helper class to build URIs for the switch browser challenge.
+ */
+object SwitchBrowserUriHelper {
+
+    private const val TAG = "SwitchBrowserUriBuilder"
+
+    /**
+     * Build the process uri for the switch browser challenge.
+     *
+     * @param brokerRedirectUri The broker redirect uri containing the switch browser code and action URI.
+     * e.g. msauth://com.microsoft.identity.client/your-redirect-uri?code=your-switch-browser-code&action_uri=your-action-uri
+     *
+     * @return The process uri constructed from the broker redirect uri.
+     * e.g. your-action-uri?code=your-switch-browser-code
+     */
+    fun buildProcessUri(brokerRedirectUri: Uri): Uri? {
+        val methodTag = "$TAG:buildProcessUri"
+        // Get the session token from the broker redirect uri.
+        val sessionToken = brokerRedirectUri.getQueryParameter(
```

Comment:
On the server side, we've moved towards calling this a `purpose token` or more specifically a `SwitchBrowserPurposeToken` as opposed to a session token. I suggest you do the same. This is just a terminology change and has no functional impact.

Reply:
From client perspective it is just a code, so you could also just name the variable as code. I guess any naming for the var is fine.

Reply:
renamed to gheneric name